### PR TITLE
asterisk: 13.38.2 -> 13.38.3, 16.17.0 -> 16.19.1, 17.9.3 -> 17.9.4, 18.3.0 -> 18.5.1

### LINKED ISSUE #124607: Vulnerability roundup 102: asterisk-13.38.2: 1 advisory [7.5]
[search](https://search.nix.gsc.io/?q=asterisk&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=asterisk+in%3Apath&type=Code)

* [ ] [CVE-2021-26712](https://nvd.nist.gov/vuln/detail/CVE-2021-26712) CVSSv3=7.5 (nixos-20.09, nixos-unstable)

Scanned versions: nixos-20.09: 76ed24ceab9; nixos-unstable: ea7d4aa9b82.

Cc @DerTim1
Cc @auntieNeo
Cc @yorickvp

### PR DESCRIPTION
**Pull request by yorickvP**
###### Motivation for this change

- fixes https://github.com/NixOS/nixpkgs/issues/124607
- wrote an update script to make this easier in the future

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yorickvP** [comment]
@ofborg build asterisk_13 asterisk_16 asterisk_17 asterisk_18

File: pkgs/servers/asterisk/default.nix [hunk 1]
```diff
@@ -90,7 +90,16 @@ let
...
   };
 
-in rec {
+  # auto-generated by update.py
+  versions = lib.mapAttrs (_: {version, sha256}: common {
+    inherit version sha256;
+    externals = {
+      "externals_cache/pjproject-2.10.tar.bz2" = pjproject_2_10;
+      "addons/mp3" = mp3-202;
+    };
```
Comment (risicle) [inline pkgs/servers/asterisk/default.nix:99]: I suppose if we needed it we'd squeeze in an `extraPatches` arg here :point_up:  which could conditionally set itself based on the `version`.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133009` at 823ac164 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>asterisk (asterisk-lts ,asterisk-stable ,asterisk_18)</li>
    <li>asterisk-module-sccp</li>
    <li>asterisk_13</li>
    <li>asterisk_16</li>
    <li>asterisk_17</li>
  </ul>
</details>

**Comment by risicle** [comment]
`nixpkgs-review` happy for me, nixos x86_64

**Comment by lukegb** [comment]
Any interest in making `update.py` work as a `passthru.updateScript`?

**Comment by yorickvP** [comment]
> Any interest in making `update.py` work as a `passthru.updateScript`?

I'll look into it.

**Pull request merged**